packages/fast-components/src/button/button.ts

```diff
@@ -0,0 +1,126 @@
+import { attr, FastElement, html, ref } from "@microsoft/fast-element";
+
+export type ButtonAppearance =
+    | "accent"
+    | "lightweight"
+    | "neutral"
+    | "outline"
+    | "stealth";
+
+export const buttonTemplate = html<Button>`
+    <button
+        class="control"
+        autofocus=${x => x.autofocus}
+        disabled=${x => x.disabled}
+        form=${x => x.form}
+        formaction=${x => x.formaction}
+        formenctype=${x => x.formenctype}
+        formmethod=${x => x.formmethod}
+        formnovalidate=${x => x.formnovalidate}
+        formtarget=${x => x.formtarget}
+        name=${x => x.name}
+        type=${x => x.type}
+        value=${x => x.value}
+    >
+        <span
+            part="start"
+            ${ref("startContainer")}
+        >
+            <slot
+                name="start"
+                ${ref("start")}
+                @slotchange=${x => x.handleStartContentChange()}
+            ></slot>
+        </span>
+        <span class="content" part="content">
+            <slot></slot>
+        </span>
+        <span
+            part="end"
+            ${ref("endContainer")}
+        >
+            <slot
+                name="end"
+                ${ref("end")}
+                @slotchange=${x => x.handleEndContentChange()}
+            ></slot>
+        </span>
+    </button>
+`;
+
+export class Button extends FastElement {
+    @attr
```

Comment:
Buttons are technically form-associated elements - spec has details on implementation with that in mind.

Reply:
Sounds good, I likely looked right past that. I’ll update.